treeherder/etl/buildapi.py

```diff
@@ -144,6 +153,16 @@ def transform(self, data, filter_to_project=None, filter_to_revision=None,
             if filter_to_revision and filter_to_revision != resultset['revision']:
                 continue
 
+            # We record the id here rather than at the start of the loop, since we
+            # must not count jobs whose revisions were not yet imported as processed,
+            # or we'll never process them once we've ingested their associated revision.
+            job_ids_seen_now.add(build['id'])
```

Comment:
> same here as below

The problem with this one, is that we'd end up caching jobs as "seen" even though we'd skipped the job since its revision wasn't known yet. Therefore we'd never import it.